llvm/include/llvm/CASObjectFormats/CASObjectReader.h

```diff
@@ -55,6 +59,8 @@ struct CASBlock {
   uint64_t AlignmentOffset;
   Optional<StringRef> Content;
   CASSectionRef SectionRef;
+  llvm::cas::CASID ID;
```

Comment:
I will call this something like `BlockContentID` or similar to emphasize that this is an ID to capture the content of the block and we can compare this content ID to see if the data in the block are the same or not.